### ISSUE TITLE
Howto use lozad with videos?

### ISSUE THREAD

**Issue by obendev** [ApoorvSaxena/lozad.js#127]
Hey!
I've included lozad.js with the cdn `<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/lozad/dist/lozad.min.js"></script>` and I use this code for inserting my video:

```
<video class="lozad" autoplay muted loop playsinline poster="488x488.png">
    <source data-src="files/videos/test.mp4" type="video/mp4">
</video>
```

If I then paste

```
const observer = lozad();
observer.observe();
```

in the console it adds the `data-loaded="true"` to the `<video>` but nothing happens.
Am I doing something wrong?
I thought it would convert `data-src` to `src`

**Comment by TiKevin83** [comment]
Have you tried doing class="lozad" on the <source> element?  Lozad looks for things with the class specified and changes data-src to src on that specific element.

**Comment by lkdvc** [comment]
Yes, I already tried `<source class="lozad" data-src="files/videos/test.mp4" type="video/mp4"`

However I'm now using an [other script from Google](https://developers.google.com/web/fundamentals/performance/lazy-loading-guidance/images-and-video/#for_video_acting_as_an_animated_gif_replacement) now because I only need it for videos anways.
This can be closed. 👍

**Comment by ApoorvSaxena** [comment]
@obendev this should be fixed in `Lozad.js`

**Comment by Levdbas** [comment]
Hi @ApoorvSaxena , any word when this is going to be adressed? :)

**Comment by ApoorvSaxena** [comment]
@Levdbas can you share a working codepen implementation along with PR & test cases. Solution looks fine to me.

**Comment by Levdbas** [comment]
Hi @ApoorvSaxena ,

Completely new to the PR/contributing part to public repo's. I will try to setup a PR but first I need to read a bit about it first. In the meantime, check out this codepen example: https://codepen.io/anon/pen/LXEZvw

updated code from last time: 

`if (element.nodeName.toLowerCase() === 'video' && !element.getAttribute('data-src')) {
      if (element.children) {
        const childs = element.children
        for (let i = 0; i <= childs.length - 1; i++) {
          childs[i].src = childs[i].getAttribute('data-src')
        }
        element.load()
      }
    }`

**Comment by ApoorvSaxena** [comment]
@Levdbas I'll be glad to wait for your PR, and guide you if needed to make your first PR to this repo.

As far as code is considered, it looks fine. Will go through the documentation of `<video>` element myself as well

**Comment by ApoorvSaxena** [comment]
this has been fixed in `v1.8.0`, thanks to @Levdbas
